src/Runner.Worker/ActionRunner.cs

```diff
@@ -179,17 +179,15 @@ public async Task RunAsync()
             ExecutionContext.Debug("Loading env");
             var environment = new Dictionary<String, String>(VarUtil.EnvironmentVariableKeyComparer);
 
-            // Apply environment set using ##[set-env] first since these are job level env
-            foreach (var env in ExecutionContext.EnvironmentVariables)
+#if OS_WINDOWS
+            var envContext = ExecutionContext.ExpressionValues["env"] as DictionaryContextData;
+#else
+            var envContext = ExecutionContext.ExpressionValues["env"] as CaseSensitiveDictionaryContextData;
```

Comment:
is env on the runner side case sensitive today on linux? i thought expressions always are case insentive dictionaries

Reply:
could go either way, but inconsistent for env defined above job level if case sensitive within job

Reply:
i wonder if should be case sensitive all the way up, and deserialize into the correct type. Runner would need to reconstruct case-insensitive on windows

Reply:
i made it case sensitive because we was trying to put machine env into the `env` context, now I am thinking maybe we just make it case-insensitive and lines up with job/workflow env.